nixos/doc/manual/development/option-types.section.md

```diff
@@ -211,9 +211,24 @@ merging is handled.
     definitions cannot be merged. The regular expression is processed
     using `builtins.match`.
 
-## Submodule types {#sec-option-types-submodule}
+## Record and module types {#sec-option-types-submodule}
```

Comment:
I feel like `record` is closer to `attrTag`, `listOf` and `attrsOf`, but that might be overly theoretical of me.

Reply:
I agree records are technically dissimilar to submodules, but I think the usage of modules and records _is_ similar.

I interpret this section of the docs as laying out the available types for scenarios involving sub-options, and scoped/nested config.